### DISCUSSION TITLE
Support for Spaces (aka Desktops)

### DISCUSSION THREAD

**Discussion by rxhanson** [rxhanson/Rectangle#402] [Ideas]
This comes up a lot, so I figured I'd leave an open issue for it.

Apple doesn't have a public API available for Spaces. There's only a private API. Amethyst and SizeUp both use sort of hacks to do this, and last time I tried it in SizeUp it didn't work for me. The choice not to use private APIs and hacks in Rectangle is the choice of more stability over more features. It's also a decent amount of code to add in (and maintain).

If Apple decided to roll out a public API for interacting with Spaces, I wouldn't hesitate to add this in.

**Comment by swswsws583** [comment]
I just tried SizeUp and it worked for me (MacOS 12.4), but sadly both Rectangle and Rectangle Pro didn't. I could see the cursor jumps to the top of the window as if it's holding it while the Desktop moves in the background. Then the cursor returns to its original position. I wonder if you can do that with Rectangle. Thanks.

**Reply by rxhanson** [reply]
Thanks for the feedback. I've updated my initial comment here. What this comes down to is that I don't have confidence that it will work for everyone, and I prioritize features that I can properly support. If anyone wants to add this feature to Rectangle as a Terminal command config with the disclaimer that it's not going to work for everyone, I'm fine merging a good PR for it.

**Comment by bt** [comment]
Given this is a discussion thread, I just wanted to leave my 2c.

This is actually the only feature that's stopping me from moving from SizeUp to Rectangle (even Pro).

As I am often using my MacBook Pro on the road, it's very common for me to move windows between spaces (desktops). I don't want to continue using SizeUp as it seems like it's a dead project as there hasn't been any updates for a long time, not even an Native Apple build. It's the only application I have left that's running in Rosetta (Intel) mode.

I've been trialling Rectangle Pro and I just came across the fact that I can't move windows to different spaces using keyboard shortcuts, so now I'm back to SizeUp unfortunately.

Would there be a compromise if you had enough interest from people that you'd add this feature into Rectangle Pro?

**Reply by rxhanson** [reply]
No promises, but I'll give it another look when I'm back on a new feature sprint on Rectangle Pro.

**Comment by 2016bgeyer** [comment]
As @rxhanson mentioned, people could fork Rectangle and add support in for spaces themselves.  There are currently 579 forks of Rectangle that I see.  Does anyone know if any of those have added support for this in?

The code from Amethyst which has the feature added in manually is detailed here:
https://superuser.com/a/1532947

**Reply by andrei8782** [reply]
Good shout! Did you end up finding a fork that has added support for it?

**Reply by 2016bgeyer** [reply]
Nope. I'm currently just using USB Overdrive to make custom macros for extra the buttons on my mouse and just using those to navigate across desktops.
